src/sdk.ts

```diff
@@ -237,8 +238,15 @@ export default class CegaEvmSDK {
     overrides: TxOverrides = {},
   ): Promise<ethers.providers.TransactionResponse> {
     const cegaState = await this.loadCegaStateContract();
+    const gasLimit = await getEstimatedGasLimit(
```

Comment:
is there a way to implement this using a pre-function hook? so that we add gas estimates automatically to every new function?

i.e. when calling `cegaState.addOracle`, it will automatically estimate gas limit with the input parameters

that way we don't have to write the same function so many times across every one of the functions?

Reply:
@vsai Good point! I had the same thought while implementing it to the SDK functions. However while a pre-function hook could save us from writing repetitive code, I thought it might add complexity. Either way, we'll need to add at least a line of code to wrap it to the contract call. LMK if you have any ideas in mind.